app/feature/feature-chat/src/main/kotlin/com/hedvig/android/feature/chat/ui/ChatDestination.kt

```diff
@@ -110,9 +113,10 @@ private fun ChatScreen(
           .consumeWindowInsets(PaddingValues(top = topAppBarHeight)),
         propagateMinConstraints = true,
       ) {
+        val loadingIndicator = remember { movableContentOf { HedvigFullScreenCenterAlignedProgress() } }
```

Comment:
The alternative to this here would be to do
```
        when (uiState) {
          ChatUiState.Initializing -> {
            HedvigFullScreenCenterAlignedProgress()
          }

          is ChatUiState.Loaded -> {
	@@ -128,6 +132,12 @@ private fun ChatScreen(
              onSendMedia = onSendMedia,
              onFetchMoreMessages = onFetchMoreMessages,
            )
            val stillLoadingInitialMessages = uiState.messages.isEmpty() &&
              (uiState.fetchMoreMessagesUiState is ChatUiState.Loaded.FetchMoreMessagesUiState.StillInitializing ||
                uiState.fetchMoreMessagesUiState is ChatUiState.Loaded.FetchMoreMessagesUiState.FetchingMore)
            if (stillLoadingInitialMessages) {
              HedvigFullScreenCenterAlignedProgress()
            }
          }
        }
```

Without movableContentOf, since we call `HedvigFullScreenCenterAlignedProgress` in two different places, what would happen before is that if the loading indicators were already mid-animation (they always are of course) then changing from Initialized to Loaded it'd mean a *new* `HedvigFullScreenCenterAlignedProgress` would show on the screen so the loading indicator would start over.

With movableContentOf, the same composable is placed on the UI in both of those cases, and the animation just continues playing from where it was already called, despite the fact that it's called from two separate parts of the UI.

This was pretty cool tbh, I hadn't found much use of movableContentOf before but here it fits perfectly.

Reply:
Nice!